packages/@react-aria/datepicker/src/useDateSegment.ts

```diff
@@ -93,8 +93,12 @@ export function useDateSegment(segment: DateSegment, state: DateFieldState, ref:
   let parser = useMemo(() => new NumberParser(locale, {maximumFractionDigits: 0}), [locale]);
 
   let backspace = () => {
+    if (segment.text === segment.placeholder) {
+      focusManager.focusPrevious();
+    }
     if (parser.isValidPartialNumber(segment.text) && !state.isReadOnly && !segment.isPlaceholder) {
       let newValue = segment.text.slice(0, -1);
+      newValue = newValue === '0' ? '' : newValue;
       let parsed = parser.parse(newValue);
```

Comment:
Did some digging and I think we'll need to change this bit of logic to something like
```suggestion
      let newValue = segment.text.slice(0, -1);
      let parsed = parser.parse(newValue);
      newValue = parsed === 0 ? '' : newValue;
```
The reason for this is that the segment text is not always to be [Arabic numerals](https://en.wikipedia.org/wiki/Arabic_numerals) (e.g. 0, 1, 2, 3 etc). If you go to [this story](https://reactspectrum.blob.core.windows.net/reactspectrum/b3cdfcf5127ebb8beaecc8b3b0248e9ac289a298/storybook/index.html?path=/story/date-and-time-datepicker--date-time-value&providerSwitcher-express=false) locally and change the locale in the story to "Arabic (Egypt)" and log out what newValue is, you'll notice that it will return values like `٤`. If you then switch your keyboard into an Arabic format so pressing `1` would return `١`, you would then see the bug still exists where the focus auto advances after clearing the minutes time field. If we parse the value first, we can get around this issue 

Lemme know if that makes sense or if you need help reproducing !

Reply:
Ah, I see. I had overlooked that perspective, thanks for pointing it out! I've tried it locally and I understand the issue now.

It looks like we can also fix the bug where input becomes impossible after using backspace to clear the minutes field in cases other than Arabic numerals, like with Arabic (Egypt).

I'll be adopting the commit you suggested. Thanks!